common/persistence/sqlMetadataPersistence.go

```diff
@@ -0,0 +1,383 @@
+		`db_version = :current_db_version`
+
+	templateDeleteDomainByIdSqlQuery   = `DELETE FROM domains WHERE id = :id`
+	templateDeleteDomainByNameSqlQuery = `DELETE FROM domains WHERE name = :name`
+)
+
+func (m *sqlMetadataManager) Close() {
+	if m.db != nil {
+		m.db.Close()
+	}
+}
+
+func gobSerialize(x interface{}) ([]byte, error) {
+	b := bytes.Buffer{}
+	e := gob.NewEncoder(&b)
+	err := e.Encode(x)
+	if err != nil {
+		return nil, err
+	}
+	return b.Bytes(), nil
+}
+
+func gobDeserialize(a []byte, x interface{}) error {
+	b := bytes.NewBuffer(a)
+	d := gob.NewDecoder(b)
+	err := d.Decode(x)
+	return err
+}
+
+func (m *sqlMetadataManager) CreateDomain(request *CreateDomainRequest) (*CreateDomainResponse, error) {
+	tx, err := m.db.Beginx()
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	// Disallow creating more than one domain with the same name, even if the UUID is different.
+	_, err = m.GetDomain(&GetDomainRequest{Name: request.Info.Name})
+	switch err.(type) {
+	case *workflow.EntityNotExistsError:
+		// Domain does not already exist. Create it.
+
+		// Encode request.ReplicationConfig.Clusters
+		clusters, err := gobSerialize(serializeClusterConfigs(request.ReplicationConfig.Clusters))
+		if err != nil {
+			return nil, &workflow.InternalServiceError{
+				Message: fmt.Sprintf("CreateDomain operation failed. Failed to encode ReplicationConfig.Clusters. Value: %v", request.ReplicationConfig.Clusters),
+			}
+		}
+
+		if _, err := tx.NamedExec(templateCreateDomainSqlQuery, &FlatCreateDomainRequest{
+			DomainInfo:   *(request.Info),
+			DomainConfig: *(request.Config),
+			// TODO Extracting the fields from DomainReplicationConfig since we don't currently support
+			// TODO DomainReplicationConfig.Clusters
+
+			//DomainReplicationConfig: *(request.ReplicationConfig),
+			ActiveClusterName: request.ReplicationConfig.ActiveClusterName,
+			Clusters:          clusters,
+
+			// TODO
+
+			IsGlobalDomain:  request.IsGlobalDomain,
+			ConfigVersion:   request.ConfigVersion,
+			FailoverVersion: request.FailoverVersion,
+		}); err != nil {
+			print(err.Error())
+			return nil, &workflow.InternalServiceError{
+				Message: fmt.Sprintf("CreateDomain operation failed. Inserting into domains table. Error: %v", err),
+			}
+		}
+
+		tx.Commit()
+		return &CreateDomainResponse{ID: request.Info.ID}, nil
+
+	case nil:
+		// The domain already exists.
+		return nil, &workflow.DomainAlreadyExistsError{
+			Message: fmt.Sprintf("Domain already exists.  DomainId: %v", request.Info.ID),
+		}
+
+	default:
+		print("default")
+		print(err.Error())
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf(
+				"CreateDomain operation failed. Could not check if domain already existed. Error: %v", err),
+		}
+	}
+}
+
+func (m *sqlMetadataManager) GetDomain(request *GetDomainRequest) (*GetDomainResponse, error) {
+	// TODO cassandraMetaPersistence_test seems to lack a test case for when both ID and name are empty
```

Comment:
Can you help add this unit test? Thanks!

Reply:
TODO in next PR: add this test to v2